src/main/java/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,91 @@
+package step2;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static java.lang.Integer.parseInt;
+
+public class StringAddCalculator {
+    public static final String REGEX = ",|:";
+    public static final String CUSTOM_REGEX = "//(.)\n(.*)";
+    public static final int DELIMITER_GROUP = 1;
+    public static final int TEXT_GROUP = 2;
+
+    private static final Pattern CUSTOM_PATTERN = Pattern.compile(CUSTOM_REGEX);
+
+    public static int splitAndSum(String text) {
+        if (text == null || text.isEmpty()) {
+            return 0;
+        }
+
+        return sum(toInts(split(text)));
+    }
+
+    public static int sum(List<Integer> numbers) {
+        int result = 0;
+
+        for (int number : numbers) {
+            result += number;
+        }
+
+        return result;
+    }
+
+    public static List<Integer> toInts(String[] values) {
+        List<Integer> numbers = new ArrayList<>();
+
+        for (String value : values) {
+            int number = getNumber(value);
+            numbers.add(number);
+        }
+
+        return numbers;
+    }
+
+    private static int getNumber(String value) {
+        int number = parseInt(value);
+
+        if (isNegative(number)) {
+            throw new RuntimeException("음수는 허용되지 않습니다.");
+        }
+
+        return number;
+    }
+
+    private static boolean isNegative(int number) {
+        return number < 0;
+    }
```

Comment:
```suggestion
    private static void validateNegativeNumber(int number) {
        if (number < 0) {
            throw new RuntimeException("음수는 허용되지 않습니다.");
        }
    }
```
(반영하지 않으셔도 됩니다.) 메서드를 하나의 기능이라고 바라보는 관점에서 음수임을 판별하는 메서드를 분리할 수 있겠지만, 음수인 것을 vaildation하는 메서드를 분리한다라는 관점으로 분리해볼 수도 있을 것 같아요!